### ISSUE TITLE
Word_idx and Token_idx

### ISSUE THREAD

**Issue by soham-joshi** [nickgkan/beauty_detr#17]
Hi @ayushjain1144 
Could you clarify what the numpy arrays word_idx and token_idx in train_dist_mod.py are created for?
reference: [line 205](https://github.com/nickgkan/butd_detr/blob/main/train_dist_mod.py#L205)

Thanks!

**Comment by soham-joshi** [comment]
I had a query, what does end_points['last_sem_cls_scores'] (dim=(B, 256, 256) ) in the same function represent?

**Comment by nickgkan** [comment]
We contrast each token in the sentence with each query. The predictions for one piece of text are the predictions of the queries, with confidence equal to the similarity of the projected queries and respective tokens.

For object detection prompts, we collect class scores and predictions by looping over the classes (word_idx) and aggregating the scores of the corresponding tokens (token_idx).

Then we use these to fill the semantic scores for the last prediction head, which are fed to the evaluator.

**Comment by soham-joshi** [comment]
Okay
Moreover, what do the second and third dimensions represent in  end_points['last_sem_cls_scores']? Do they represent logits over a set of (256?) classes?

**Comment by ayushjain1144** [comment]
the 2nd dimension represents the number of queries, and the third dimension is logits over the 256 tokens in a sentence span. Each query predicts a distribution over the sentence span.

**Comment by soham-joshi** [comment]
Thanks for the responses @ayushjain1144 @nickgkan !
